vsphere/resource_vsphere_nas_datastore_test.go

```diff
@@ -0,0 +1,455 @@
+package vsphere
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccResourceVSphereNasDatastore(t *testing.T) {
+	var tp *testing.T
+	testAccResourceVSphereNasDatastoreCases := []struct {
+		name     string
+		testCase resource.TestCase
+	}{
+		{
+			"basic",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccResourceVSphereNasDatastorePreCheck(tp)
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereNasDatastoreExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereNasDatastoreConfigBasic(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereNasDatastoreExists(true),
+						),
+					},
+				},
+			},
+		},
+		{
+			"multi-host",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+					testAccResourceVSphereNasDatastorePreCheck(tp)
+					testAccSkipIfEsxi(tp)
+				},
+				Providers: testAccProviders,
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereNasDatastoreConfigMultiHost(),
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereNasDatastoreExists(true),
```

Comment:
It doesn't look like it but I'd just like to confirm that there isn't any other variable to check if it's basic/multi-host?

Reply:
`Read` picks this up reasonably well (the host ID list is updated on refresh, and if there's a mismatch the test will fail with a non-empty plan)... we could probably have a harder check, but writing the check function might be a little overly complex as we would have to derive the host IDs from the host names in the environment variables (`VSPHERE_ESXI_HOST`, `VSPHERE_ESXI_HOST2`, and `VSPHERE_ESXI_HOST3`). I think this should be pretty good unless there's some sort of serious edge case that comes up.